src/rust/src/x509/certificate.rs

```diff
@@ -126,6 +126,18 @@ impl Certificate {
             py,
             &asn1::write_single(&self.raw.borrow_value().tbs_cert.spki),
         );
+        let backend = py.import("cryptography.hazmat.backends.openssl.backend")?;
+        let lib = backend.getattr("backend")?.getattr("_lib")?;
+        let has_pss = lib
+            .getattr("Cryptography_HAS_EVP_PKEY_RSA_PSS")?
+            .extract::<u8>()?;
+        if self.raw.borrow_value().signature_alg.oid == *x509::oid::RSASSA_PSS_OID && has_pss == 0 {
```

Comment:
Err, is it really the signature alg we need to check ehre? Surely it's the spki algorithm?

Reply:
Ugh, quite correct.